extensions/oidc/runtime/src/main/java/io/quarkus/oidc/runtime/DefaultTokenStateManager.java

```diff
@@ -59,21 +65,26 @@ public Uni<String> createTokenState(RoutingContext routingContext, OidcTenantCon
                 }
             }
         }
-        return Uni.createFrom().item(sb.toString());
+        String state = encryptAll ? encryptToken(sb.toString(), routingContext, oidcConfig) : sb.toString();
```

Comment:
Are we 100% sure that there will never be a pipe character in the token?

Reply:
Hi @stuartwdouglas Yes, it is safe, we used to have problems when `_` was used as a separator, https://github.com/quarkusio/quarkus/pull/9519, since `_` is a valid base64url character but it all stabilized after a pipe character became a separator.